CharlieBackend.Api.UnitTes/ScheduleServiceTests.cs

```diff
@@ -409,5 +414,438 @@ public async Task CreateScheduleAsync_NullRequest_ShouldReturnValidationError()
+
+            var scheduleService = new ScheduleService(_unitOfWorkMock.Object, _mapper, _scheduledEventFactory);
+
+            //Act
+            var result = await scheduleService.GetEventOccurrenceByIdAsync(existingId);
+
+            //Assert
+            result.Should().NotBeNull();
+            result.Data.Should().BeEquivalentTo(_mapper.Map<EventOccurrenceDTO>(validEventOccurrence));
+        }
+
+        [Fact]
+        public async Task GetEventOccurrenceByIdAsync_NonExistingId_ShouldReturnNotFound()
+        {
+            //Arrange
+            var scheduleService = new ScheduleService(_unitOfWorkMock.Object, _mapper, _scheduledEventFactory);
+
+            //Act
+            var result = await scheduleService.GetEventOccurrenceByIdAsync(nonExistentId);
+
+            //Assert
+            result.Should().NotBeNull();
+            result.Error.Code.Should().BeEquivalentTo(ErrorCode.NotFound);
+        }
+
+        [Fact]
+        public async Task GetEventsFiltered_ValidScheduledEventFilterRequest_ShouldReturnFilteredListOfScheduleEventDTO()
+        {
+            //Arrange
+            var validRequest = new ScheduledEventFilterRequestDTO
+            {
+                MentorID = existentMentorId,
+                GroupID = existentGroupId
+            };
+
+            var expectedFilteredList = new List<ScheduledEventDTO>()
+            {
+            new ScheduledEventDTO
+            {
+                StudentGroupId = existentGroupId,
+                ThemeId = existentThemeId,
+                MentorId = existentMentorId,
+                EventStart = DateTime.Parse("Jan 1, 2021"),
+                EventFinish = DateTime.Parse("Feb 1, 2021"),
+                EventOccuranceId = 0,
+                Id = 0
+            } };
+
+            _scheduleRepositoryMock.Setup(x => x.GetEventsFilteredAsync(validRequest)).ReturnsAsync(new List<ScheduledEvent>()
+            {
+            new ScheduledEvent
+            {
+                StudentGroupId = existentGroupId,
+                ThemeId = existentThemeId,
+                MentorId = existentMentorId,
+                EventStart = DateTime.Parse("Jan 1, 2021"),
+                EventFinish = DateTime.Parse("Feb 1, 2021"),
+            } });
+
+            _mentorRepositoryMock.Setup(x => x.IsEntityExistAsync(existentMentorId)).ReturnsAsync(true);
+
+            _studentGroupRepositoryMock.Setup(x => x.IsEntityExistAsync(existentGroupId)).ReturnsAsync(true);
+
+            Initialize(validScheduleDTO);
+
+            var scheduleService = new ScheduleService(_unitOfWorkMock.Object, _mapper, _scheduledEventFactory);
+
+            //Act
+            var result = await scheduleService.GetEventsFiltered(validRequest);
+
+            //Assert
+            result.Should().NotBeNull();
+            result.Data.ElementAt(0).Should().BeEquivalentTo(expectedFilteredList.ElementAt(0));
```

Comment:
```suggestion
            result.Data.First().Should().BeEquivalentTo(expectedFilteredList.First());
```
The same but without hardcoded id.